### ISSUE TITLE
Improve generated code recognition

### ISSUE THREAD

**Issue by tamasvajk** [SonarSource/sonar-csharp#905]
Swagger Codegen uses the following comment to mark generated files:

```C#
/* 
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 1.0.0
 * 
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 */
```

Could you improve generated file recognition to exclude files from analysis if there's a header comment that contains "generated by"?

**Comment by Evangelink** [comment]
Google protobuf is following a similar pattern:
`// Generated by the protocol buffer compiler.  DO NOT EDIT!`

**Comment by Evangelink** [comment]
Closed by #911.
